tests/framework/extensions/nodes/ec2/ec2.go

```diff
@@ -10,138 +10,174 @@ import (
+		err = ec2Client.SVC.WaitUntilInstanceRunning(&ec2.DescribeInstancesInput{
+			InstanceIds: listOfInstanceIds,
+		})
+		if err != nil {
+			return nil, nil, err
+		}
 
-	//wait until instance status is ok
-	err = ec2Client.SVC.WaitUntilInstanceStatusOk(&ec2.DescribeInstanceStatusInput{
-		InstanceIds: listOfInstanceIds,
-	})
-	if err != nil {
-		return nil, err
-	}
+		//wait until instance status is ok
+		err = ec2Client.SVC.WaitUntilInstanceStatusOk(&ec2.DescribeInstanceStatusInput{
+			InstanceIds: listOfInstanceIds,
+		})
+		if err != nil {
+			return nil, nil, err
+		}
 
-	// describe instance to get attributes=
-	describe, err := ec2Client.SVC.DescribeInstances(&ec2.DescribeInstancesInput{
-		InstanceIds: listOfInstanceIds,
-	})
-	if err != nil {
-		return nil, err
-	}
+		// describe instance to get attributes=
+		describe, err := ec2Client.SVC.DescribeInstances(&ec2.DescribeInstancesInput{
+			InstanceIds: listOfInstanceIds,
+		})
+		if err != nil {
+			return nil, nil, err
+		}
 
-	readyInstances := describe.Reservations[0].Instances
+		readyInstances := describe.Reservations[0].Instances
 
-	sshKey, err := nodes.GetSSHKey(ec2Client.Config.AWSSSHKeyName)
-	if err != nil {
-		return nil, err
-	}
+		sshKey, err := nodes.GetSSHKey(config.AWSSSHKeyName)
+		if err != nil {
+			return nil, nil, err
+		}
 
-	var ec2Nodes []*nodes.Node
+		for _, readyInstance := range readyInstances {
+
+			if multiconfig {
+				if config.AWSAMI == ec2Client.ClientConfig.AWSEC2Config[0].AWSAMI &&
+					config.AWSUser == ec2Client.ClientConfig.AWSEC2Config[0].AWSUser {
+					ec2Node := &nodes.Node{
+						NodeID:          *readyInstance.InstanceId,
+						PublicIPAddress: *readyInstance.PublicIpAddress,
+						SSHUser:         config.AWSUser,
+						SSHKey:          sshKey,
+					}
+					ec2Nodes = append(ec2Nodes, ec2Node)
+				}
+
+				if config.AWSAMI == ec2Client.ClientConfig.AWSEC2Config[1].AWSAMI &&
+					config.AWSUser == ec2Client.ClientConfig.AWSEC2Config[1].AWSUser {
+					ec2Node2 := &nodes.Node{
+						NodeID:          *readyInstance.InstanceId,
+						PublicIPAddress: *readyInstance.PublicIpAddress,
+						SSHUser:         config.AWSUser,
+						SSHKey:          sshKey,
+					}
+					ec2Nodes2 = append(ec2Nodes2, ec2Node2)
+				}
+			}
+			if !multiconfig {
+				ec2Node := &nodes.Node{
+					NodeID:          *readyInstance.InstanceId,
+					PublicIPAddress: *readyInstance.PublicIpAddress,
+					SSHUser:         config.AWSUser,
+					SSHKey:          sshKey,
+				}
+				ec2Nodes = append(ec2Nodes, ec2Node)
+				ec2Nodes2 = nil
+			}
+		}
+
+		client.Session.RegisterCleanupFunc(func() error {
+			return DeleteNodes(client, ec2Nodes, ec2Nodes2, multiconfig)
+		})
 
-	for _, readyInstance := range readyInstances {
-		ec2Node := &nodes.Node{
-			NodeID:          *readyInstance.InstanceId,
-			PublicIPAddress: *readyInstance.PublicIpAddress,
-			SSHUser:         ec2Client.Config.AWSUser,
-			SSHKey:          sshKey,
+		if !multiconfig {
```

Comment:
unsure, but is this needed?

Reply:
Yeah I agree, do we even need this?

Reply:
Yes, because it is making the distinction between tests that only have 1 instance configuration vs 2 configurations